### DISCUSSION TITLE
Question about video textures

### DISCUSSION THREAD

**Discussion by webdevelopment77** [google/model-viewer#4424] [Q&A]
Hello, 

I have a question about textures in modelviewer.

I know how to set one videotexture in model - 

`const material = modelViewerAnimated.model.materials[3];
  
  const {baseColorTexture} = material.pbrMetallicRoughness;
  
  baseColorTexture.setTexture(videoTexture);`

But how can I set two video textures in one model?

If I write like this:

`const material = modelViewerAnimated.model.materials[3];
  
  const {baseColorTexture} = material.pbrMetallicRoughness;
  
  baseColorTexture.setTexture(videoTexture);
  
  const material2 = modelViewerAnimated.model.materials[5];
  
  const {baseColorTexture2} = material2.pbrMetallicRoughness; 
  
  baseColorTexture2.setTexture(videoTexture);`

Than I have a message error - 

Uncaught TypeError: Cannot read properties of undefined (reading 'setTexture')
    at HTMLElement.<anonymous> ((index):294:21)
    at model-viewer.js:56453:30

Am I understand correct that I can't write - baseColorTexture2. I should write baseColorTexture?

**Comment by elalish** [comment]
Correct, there's no such thing as `baseColorTexture2`, only `baseColorTexture`. If it's from a different `material`, it'll be a different object (generally).

**Reply by dzhakhangirov** [reply]
Is it possible to use two video textures for one glb model?

**Reply by elalish** [reply]
yes, if there are two textures in the model.

**Comment by dzhakhangirov** [comment]
Thank you, [elalish](https://github.com/elalish)! [webdevelopment77](https://github.com/webdevelopment77), this code seems to work:

  const material = modelViewerAnimated.model.materials[10];
  const material2 = modelViewerAnimated.model.materials[19];
  
  const baseColorTexture = material.pbrMetallicRoughness.baseColorTexture;  
  const baseColorTexture2 = material2.pbrMetallicRoughness.baseColorTexture;  
     
  baseColorTexture.setTexture(videoTexture);
  baseColorTexture2.setTexture(videoTexture2);

**Comment by webdevelopment77** [comment]
const material = modelViewerAnimated.model.materials[10];
const material2 = modelViewerAnimated.model.materials[19];

const baseColorTexture = material.pbrMetallicRoughness.baseColorTexture;
const baseColorTexture2 = material2.pbrMetallicRoughness.baseColorTexture;

baseColorTexture.setTexture(videoTexture);
baseColorTexture2.setTexture(videoTexture2);

This code works ) Thank you )
